mypy/stubtest.py

```diff
@@ -1679,16 +1680,16 @@ def get_importable_stdlib_modules() -> set[str]:
         all_stdlib_modules = sys.stdlib_module_names
     else:
         all_stdlib_modules = set(sys.builtin_module_names)
-        python_exe_dir = Path(sys.executable).parent
+        modules_by_finder: defaultdict[importlib.machinery.FileFinder, set[str]] = defaultdict(set)
         for m in pkgutil.iter_modules():
-            finder = m.module_finder
-            if isinstance(finder, importlib.machinery.FileFinder):
-                finder_path = Path(finder.path)
-                if (
-                    python_exe_dir in finder_path.parents
-                    and "site-packages" not in finder_path.parts
-                ):
-                    all_stdlib_modules.add(m.name)
+            if isinstance(m.module_finder, importlib.machinery.FileFinder):
+                modules_by_finder[m.module_finder].add(m.name)
+        for finder, module_group in modules_by_finder.items():
+            if (
+                "site-packages" not in Path(finder.path).parents
+                and {"json", "_json", "_queue"} & module_group
```

Comment:
Why are these modules special?

Reply:
I added a comment in bf9fdad2ae7542fb85ae9ffba2b488924240769a. If there was a sure-fire way of finding out where the stdlib modules are located on any Python build, we could filter according to the `.path` attribute of the finder. There might be a sure-fire way of doing so using `sysconfig`, not sure.

Reply:
Maybe these?

```
>>> sysconfig.get_path("stdlib")
'/Users/jelle/.pyenv/versions/3.11.1/lib/python3.11'
>>> sysconfig.get_path("platstdlib")
'/Users/jelle/py/venvs/py311/lib/python3.11'
```

Reply:
> Maybe these?

They don't do the right thing on Windows, unfortunately:

```pycon
>>> import sysconfig, pkgutil
>>> sysconfig.get_path("stdlib")
'C:\\Users\\alexw\\AppData\\Local\\Programs\\Python\\Python311\\Lib'
>>> sysconfig.get_path("platstdlib")
'C:\\Users\\alexw\\AppData\\Local\\Programs\\Python\\Python311\\Lib'
>>> m = next(m for m in pkgutil.iter_modules() if m.name == "_queue")
>>> m.module_finder.path
'C:\\Users\\alexw\\AppData\\Local\\Programs\\Python\\Python311\\DLLs'
```

Reply:
`os.path.join(sysconfig.get_config_var("BINDIR"), sysconfig.get_config_va
r("platlibdir"))` might be the incantation I'm looking for...

Reply:
> `os.path.join(sysconfig.get_config_var("BINDIR"), sysconfig.get_config_var("platlibdir"))` might be the incantation I'm looking for...

no, that doesn't work on linux.

There's a StackOverflow answer here using `distutils.sysconfig.get_python_lib`, but it's nontrivial, and `distutils` is all deprecated anyway: https://stackoverflow.com/a/8992937/13990016. Not sure if it's worth it.